Convert qbatch_norm to new operator registration API.

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #36536 Functionalize qconcat and register with new registration API
* #36535 Convert qclamp to new operator registration API.
* **#36534 Convert qbatch_norm to new operator registration API.**
* #36531 Update quantized README for registering operators with new API.
* #36527 Functionalize qadd and register with new registration API.
* #36520 Add a centralized TORCH_LIBRARY declaration for quantized and xnnpack ops.
* #36513 Add explicit schema for quantized conv/conv_prepack
* #36512 Port detach/detach to new registration API
* #36390 Port Resize to use new registration API.
* #36389 Update reference to RegisterOperators in error message in Convolution
* #36258 Switch to pybind11 style registration function API.

Say good bye to boilerplate.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21017672](https://our.internmc.facebook.com/intern/diff/D21017672)

ezyang committed:
Convert qbatch_norm to new operator registration API.

Say good bye to boilerplate.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Convert qbatch_norm to new operator registration API."


Say good bye to boilerplate.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

supriyar commented:
I'm updating the op name from batch_norm to batch_norm2d in #36494. Can we land this after that?

ezyang commented:
Yup

ezyang committed:
Update on "Convert qbatch_norm to new operator registration API."


Say good bye to boilerplate.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Convert qbatch_norm to new operator registration API."


Say good bye to boilerplate.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21017672](https://our.internmc.facebook.com/intern/diff/D21017672)

[ghstack-poisoned]

ezyang committed:
Update on "Convert qbatch_norm to new operator registration API."


Say good bye to boilerplate.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21017672](https://our.internmc.facebook.com/intern/diff/D21017672)

[ghstack-poisoned]

ezyang committed:
Update on "Convert qbatch_norm to new operator registration API."


Say good bye to boilerplate.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21017672](https://our.internmc.facebook.com/intern/diff/D21017672)

[ghstack-poisoned]

ezyang committed:
Update on "Convert qbatch_norm to new operator registration API."


Say good bye to boilerplate.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21017672](https://our.internmc.facebook.com/intern/diff/D21017672)

[ghstack-poisoned]

ezyang commented:
This will be rolled up into #36389
